core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -368,21 +378,21 @@ trait Endpoint[A] { self =>
  */
 object Endpoint {
 
-  type Result[A] = EndpointResult[A]
+  type Result[F[_], A] = EndpointResult[F, A]
 
   /**
    * Creates an empty [[Endpoint]] (an endpoint that never matches) for a given type.
    */
-  def empty[A]: Endpoint[A] = new Endpoint[A] {
-    final def apply(input: Input): Result[A] = EndpointResult.NotMatched
+  def empty[F[_] : Effect, A]: Endpoint[F, A] = new Endpoint[F, A] {
```

Comment:
I liked how Monix's `Iterant` provides [its builders](https://github.com/monix/monix/blob/master/monix-tail/shared/src/main/scala/monix/tail/Iterant.scala#L1998) to enable the following syntax:

```
Endpoint[IO].empty[Foo]
```

Can we adopt this here?